src/app/hooks.ts

```diff
@@ -0,0 +1,6 @@
+import { useDispatch } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+import { useSelector } from 'react-redux';
+
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
+export const useAppSelector = useSelector.withTypes<RootState>();
```

Comment:
The `useSelector.withTypes<RootState>()` is incorrect. You should use `useSelector<RootState>()` instead to correctly type the selector function.